packages/ui/src/app/pages/Forum/ForumThread.tsx

```diff
@@ -79,13 +78,9 @@ export const ForumThread = () => {
     }
 
     return (
-      <div>
-        {posts.map((post) => (
-          <div>
-            {post.id} | {post.text} | by {post.authorId}
-          </div>
-        ))}
-      </div>
+      <MainPanel ref={sideNeighborRef}>
+        <PostList threadId={thread?.id as string} />
```

Comment:
Casting like this should be done only as a last resort of enforcing types. It is better to avoid such cast to not introduce bugs. Here it will create an invalid cast when `thread` is `undefined`. This should be either handled here or on the `PostList` component. I'd go with rendering this component here only if thread is not undefined.

Reply:
Well, I've included thread condition in addition to isLoading, but in general it couldn't be undefined here - if no thread then we redirect to 404